account_invoice_blocking/migrations/10.0.1.0.1/post-migration.py

```diff
@@ -0,0 +1,74 @@
+# -*- coding: utf-8 -*-
+# Copyright 2017 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import logging
+import odoo
+
+_logger = logging.getLogger(__name__)
+
+
+def update_draft_invoice_blocked_field_value(env):
+    env.cr.execute("""
+        UPDATE account_invoice
+        SET blocked = draft_blocked
+        WHERE state = 'draft'
+    """)
+
+
+def update_open_invoice_blocked_field_value(env):
+    query = """
+        SELECT ai.id as inv_id, aml.id as aml_id, aml.blocked
+        FROM account_move_line aml
+        LEFT JOIN account_invoice ai on aml.invoice_id = ai.id
+        LEFT JOIN account_account aa on aa.id = aml.account_id
+        LEFT JOIN account_account_type aat on aat.id = aa.user_type_id
+        WHERE aat.type in %s
+        ORDER BY ai.id
+    """
+    types = ('receivable', 'payable')
+    env.cr.execute(query, [types])
+    rows = env.cr.fetchall()
+
+    current_inv_id = None
+
+    blocked_by_invoice_ids = {}
+    invoice_ids_to_block = []
+
+    for row in rows:
+        inv_id = row[0]
+
+        if inv_id != current_inv_id and current_inv_id is not None:
+            if all(blocked_by_invoice_ids[current_inv_id]):
```

Comment:
This condition will never called for the last invoice block

Reply:
I added the missing condition at the end of the loop.